pkg/rescaffold/migrate.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+	http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rescaffold
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+
+	"github.com/spf13/afero"
+	"sigs.k8s.io/kubebuilder/v3/pkg/config/store"
+	"sigs.k8s.io/kubebuilder/v3/pkg/config/store/yaml"
+	"sigs.k8s.io/kubebuilder/v3/pkg/machinery"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugin/util"
+)
+
+type MigrateOptions struct {
+	InputDir  string
+	OutputDir string
+}
+
+const DefaultOutputDir = "output-dir"
+
+func (opts *MigrateOptions) Rescaffold() error {
+	store := yaml.New(machinery.Filesystem{FS: afero.NewOsFs()})
+	if err := store.LoadFrom(opts.InputDir); err != nil {
+		log.Fatal(err)
+		os.Exit(1)
+	}
+	// create output directory
+	if err := os.MkdirAll(opts.OutputDir, os.ModePerm); err != nil {
```

Comment:
Question:
Do we need to give that much permission as `ModePerm` to the folder?

Reply:
Oh, that's a question. What about set it to 0755(the default permission of directory)? According to https://www.redhat.com/sysadmin/introduction-chmod

Reply:
As discussed, we can apply `0755`.